charts/agent-control-deployment/templates/preinstall-job-register-system-identity.yaml

```diff
@@ -111,185 +111,65 @@ spec:
                 exit 0
               fi
 
-              echo Generating the Keys...
-              TEMPORAL_FOLDER=gen-folder
-              mkdir $TEMPORAL_FOLDER
-              openssl genrsa -out "$TEMPORAL_FOLDER/key" 4096
-              openssl rsa -in "$TEMPORAL_FOLDER/key" -pubout -out "$TEMPORAL_FOLDER/pub"
-
-              echo Key generated, creating the identity...
-              if [ "${NEW_RELIC_AUTH_CLIENT_ID:-}" != "" ] && [ "${NEW_RELIC_AUTH_CLIENT_SECRET:-}" != "" ]; then
-                echo Starting with L1 System Identity...
-                ############################################################
-                # Get the L1 Access Token
-                ############################################################
-                RESPONSE_FILE=$TEMPORAL_FOLDER/response_token.json
-                for RETRY in 1 2 3; do
-                  HTTP_CODE=$(echo '{"client_id": "'$NEW_RELIC_AUTH_CLIENT_ID'", "client_secret": "'$NEW_RELIC_AUTH_CLIENT_SECRET'", "grant_type": "client_credentials"}' | tr -d $'\n' | curl \
-                    -s -S -w "%{http_code}" \
-                    -H "Content-Type: application/json" \
-                    -o "$RESPONSE_FILE" \
-                    --data-binary @- \
-                    --max-time 60 \
-                    "{{ include "newrelic-agent-control.config.endpoints.tokenRenewal" . }}")
+              echo "Secret not present, creating a new System Identity..."
 
-                  if [ $HTTP_CODE -eq 200 ]; then
-                    break
-                  fi
+              REGION={{ include "newrelic.common.region" . }}
+              echo "Authenticating with New Relic ($REGION)..."
 
-                  if [ -z "$HTTP_CODE" ] || [ "$HTTP_CODE" -eq 0 ]; then
-                    echo "Network error occurred or no HTTP response was received. Retrying ($RETRY/3)..."
-                    sleep 2
-                    continue
-                  else
-                    if jq empty "$TEMPORAL_FOLDER/response_token.json" > /dev/null 2>&1; then
-                      ERROR_MESSAGE=$(jq '.error_description // "invalid_request"' < "$TEMPORAL_FOLDER/response_token.json" | tr -d '"')
-                      echo "Error getting system identity auth token. The API endpoint returned $HTTP_CODE: $ERROR_MESSAGE. Retrying ($RETRY/3)..."
-                      sleep 2
-                    else
-                      echo -n "Error getting system identity auth token. The API endpoint returned $HTTP_CODE: " && cat "$TEMPORAL_FOLDER/response_token.json" | tr -d '\n' && echo " Retrying ($RETRY/3)..."
-                      sleep 2
-                      continue
-                    fi
-                  fi
-                done
+              ACCESS_TOKEN=""
+              for RETRY in 1 2 3; do
```

Comment:
We need a new flag in the cli I think. A `--retries` flag. The `TokenRetrieverWithCache` has a retries field. Would it simplify this loop here?

Reply:
Not a big deal. In fact, we don't have the same behaviour in place for "identity creation". So maybe it's better to let it as it is for consistency.

Reply:
I also thought on that... I think the ideal state would be to just cover everything in the CLI so we can just replace the whole script with a single call, but we are not there yet.